src/components/AppBar/AppBar.svelte

```diff
@@ -0,0 +1,99 @@
+<script lang="ts">
+	import type { Snippet } from "svelte";
+
+	interface AppBarProps {
+		/* Styling */
+		rootBackground?: string;
+		rootBorder?: string;
+		rootPadding?: string;
+		rootShadow?: string;
+		rootSpacing?: string;
+		rootGridColumns?: string;
+		rootGap?: string;
+		regionRowMain?: string;
+		regionRowHeadline?: string;
+		leadSnippetClasses?: string;
```

Comment:
Let's drop the "snippet" descriptor in props. This won't work cross platform as this is a Svelte-only concept. Instead, use semantic names for the target element itself. Like root, panel, control, lead, trail, fallback, etc.

Also, most elements should have a minimum of two style props:

- `fooBase` - houses the base styles for X element
- `fooRest` - replaces `regionX` props, for passing arbitrary additions

So for these snippet props, that would become:

```
leadBase / leadRest
centerBase / centerRest (i'm flexible on this one)
trailBase / trailRest
```